lib/src/entrypoint.dart

```diff
@@ -337,11 +356,8 @@ class Entrypoint {
       });
     } catch (e) {
       if (onlyReportSuccessOrFailure && (e is ApplicationException)) {
-        final directoryOption = root.isInMemory || root.dir == '.'
-            ? ''
-            : ' --directory ${root.dir}';
         throw ApplicationException(
-            'Resolving dependencies$suffix failed. For details run `$topLevelProgram pub ${type.toString()}$directoryOption`');
+            'Resolving dependencies$suffix failed.${forDetails()}');
```

Comment:
I don't think forDetails can be reused out of this flow. So why not move it in here.

Reply:
I'm using it also below when the resolution succeeded, but something changed from the pubspec.lock.